src/core/server.js

```diff
@@ -676,6 +676,17 @@ Server.prototype._handleResponse = function(message) {
     const result = message.result;
     const responseEvent = 'response_' + command;
 
+    if (command === 'submit' && result.engine_result === 'tefPAST_SEQ' &&
+        request.reconnectionsCount > 0
+    ) {
+      // handle case when request reached the server, but response was lost
+      // and request was automatically repeated
+      result.engine_result = 'tesSUCCESS';
```

Comment:
Although it kind of makes sense, this smells funny. I'm thinking we should undo the automatic re-request thing, make sure `Request.timeout()` works, add check for server has ledger range [here](https://github.com/ripple/ripple-lib/blob/87fdbc932f1369b8ce73478ddc9697d89d26799b/src/core/remote.js#L1312-L1318), let transactionmanager's reconnect logic do its thing, and defer other auto-re-requesting responsibility to clients